### ISSUE TITLE
CRL and OCSP usage for ICP-Brasil Certificates

### ISSUE THREAD

**Issue by DeiGratia33** [OpenBanking-Brasil/specs-seguranca#177]
I believe it would be appropriate to the Security WG discuss the enforcement of the usage of OCSP and CRL to perform validation of the certificates.

Suggestion:

The OCSP validation should be performed (first method of validation)
The CRL validation must be performed (second method of validation, acting as fallback)

**Comment by RalphBragg** [comment]
Hi,

OCSP is relied on in multiple places throughout the ecosystem and is a requirement for CAs to support as a CAB CA. Creating a backup service where we are reliant on stale CRLs (if OCSP doesn’t work why would CRLs) needs to be impact assessed by security and by legal.

If Directory relies on a CRL that is 4 hours old and a certificate HAS been revoked. Who is liable for payments made with a revoked certificate?

If Directory allows a REVOKED or compromised certificate to be UPLOADED because it is relying on cached CRL lists that are stale, who is liable?

The same questions need to be agreed by ALL participants including the TPPs and the Banks. If a Bank or TPP can not check REAL TIME the validity of a certificate it is a significant problem for Banks and TPPs. 

Doing this will introduce other areas of system risk, processes will have to be cycled to pick up new CRL files for those that use gateways that need to be side loaded. 

The certificate practice statements of ALL ICP CAs need to be reviewed to understand what happens and who is liable ‘if you do not check’ a certificate and it has been revoked and you rely on the certificate.

Thishas major policy and legal implications for the entire ecosystem including the CA’s. Happy to discuss what processes the UK took and what the impact on this ‘ticket’ really is with the ecosystem. 

In short OCSP is required and vital part of the security of this ecosystem. CRL’s are a poor substitute.

**Comment by DeiGratia33** [comment]
Just remember, OCSP does not mean just in time.
CAB/Forum
4.9.10 On‑line revocation checking requirements

**Comment by RalphBragg** [comment]
If only there was a PKI out there that would immediately reflect a status change of a software statement or an organisations presence on the directory...

**Comment by joaorodolfo** [comment]
We had discussed it into security WG. We will include it on SD FAQ.
